data-analytics/iot-nirvana/client/src/test/java/com/google/cloud/demo/iot/nirvana/client/ClientOptionsTest.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (C) 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.demo.iot.nirvana.client;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Unit tests for {@link ClientOptions}. */
+@RunWith(JUnit4.class)
+public final class ClientOptionsTest {
+
+  private String argsIncomplete[] = {
+    "-region", "us-central1-f",
+    "-pubSubTopicName", "projects/my-project/topics/my-topic",
+    "-registryName", "my-registry",
+    "-rsaCertificateFilePath", "/tmp/iot-core/rsa_cert_0.pem",
+    "-privateKey", "/tmp/iot-core/rsa_private_0_pkcs8",
+    "-cityIndex", "1"};
+
+  private String argsComplete[] = {
+    "-projectId", "my-project",
+    "-region", "us-central1-f",
+    "-pubSubTopicName", "projects/my-project/topics/my-topic",
+    "-registryName", "my-registry",
+    "-rsaCertificateFilePath", "/tmp/iot-core/rsa_cert_0.pem",
+    "-privateKey", "/tmp/iot-core/rsa_private_0_pkcs8",
+    "-cityIndex", "1"};
+
+  @Test
+  public void parse_exceptionThrownArgsEmpty() throws Exception {
```

Comment:
Checkout this pattern to assertThat a particular exception with a particular message is thrown in a unit test:
https://github.com/sabhyankar/professional-services/blob/b54c8f70b4bb895fe9da2c35d42eb879980f931e/data-analytics/dataflow-elasticsearch-indexer/ElasticIndexer/src/test/java/com/google/cloud/pso/common/ExtractKeyFnTest.java#L53

Reply:
Changed in commit c520a1aa6568a6aa0912d1bd2e40cab2810fe312